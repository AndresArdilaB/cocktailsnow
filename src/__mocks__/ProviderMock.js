import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
