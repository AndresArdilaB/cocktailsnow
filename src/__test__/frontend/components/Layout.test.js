import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import Layout from '../../../frontend/components/Layout';

describe('<Layout />', () => {
  const layout = mount(
    <ProviderMock>
      <Layout />
    </ProviderMock>,
  );

  test('Layout component render', () => {
    expect(layout.length).toEqual(1);
  });

});

describe('Layout Snapshot', () => {

  test('Testing the layout UI component', () => {
    const layout = create(
      <ProviderMock>
        <Layout />
      </ProviderMock>,
    );
    expect(layout.toJSON()).toMatchSnapshot();
  });
});
