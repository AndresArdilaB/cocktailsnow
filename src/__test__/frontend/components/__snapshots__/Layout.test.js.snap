// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshot Testing the layout UI component 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </header>
  <footer
    className="footer"
  >
    From Andres Ardila for Platzi - 2020
  </footer>
</div>
`;
